be/dtm0_log.c

```diff
@@ -0,0 +1,284 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2013-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_DTM
+#include "be/dtm0_log.h"
+#include "be/list.h"
+#include "lib/assert.h" /* M0_PRE */
+#include "lib/errno.h"  /* ENOENT */
+#include "lib/memory.h" /* M0_ALLOC */
+
+enum {
+	dl_magix = 0xdeaddeaddeaddead
+};
+
+static const struct m0_tl_descr dldesc = M0_TL_DESCR("DTM0 Log Record",
+                                                     struct m0_dtm0_log_rec,
+                                                     dlr_tlink,
+                                                     dlr_magic,
+                                                     dl_magix,
+                                                     0xbadbadbadbadbadb);
```

Comment:
This is a magic for the head of the list. It also has to be be defined. It could be something like `M0_BE_DTM_LREC_HEAD_MAGIX` or `M0_BE_DTM_LOG_MAGIX`.

Reply:
Done